### ISSUE TITLE
Option to clear browser data

### ISSUE THREAD

**Issue by vincode-io** [Ranchero-Software/NetNewsWire#3650]
Occasionally I have a site that sets cookies or other data that causes problems. Any chance we could get an option to clear all browser data from the app?

https://twitter.com/islandguybyu/status/1557150691004764161

**Comment by vincode-io** [comment]
Some ideas on how this could be done.

https://stackoverflow.com/questions/31289838/how-to-delete-wkwebview-cookies

**Comment by tylerbenson** [comment]
Thanks for posting here. Didn't know this was the right place for this. Appreciate the consideration.

**Comment by vincode-io** [comment]
It looks like our internal Safari browser and the standalone app are sharing some cookies and data. Have you tried clearing everything using the Safari item in Settings?

**Comment by vincode-io** [comment]
Thinking about this more, I don't think this would be possible for us to implement. The internal Safari browser runs in a different process than NNW. That is how it shares data with the standalone Safari browser. We wouldn't have access to that data to delete it from NNW. 

I'm closing this issue since I don't think it is actionable. If we find out different we can reopen it.

**Comment by stuartbreckenridge** [comment]
@vincode-io https://gist.github.com/insidegui/4a5de215a920885e0f36294d51263a15

Is that what we need?

**Comment by vincode-io** [comment]
No. That is for the WKWebView which we only use to render RSS articles. It is SFSafariViewController that visits sites on the web and is where cookies would be picked up.

**Comment by tylerbenson** [comment]
Are you sure those cookies are managed by Safari? Pretty sure the login state for safari isn't shared with NNW's browser.

**Comment by tylerbenson** [comment]
Darn. Sounds like SFSafariViewController might not give access to this as you mentioned. Seems an unfortunate oversight.

**Comment by tylerbenson** [comment]
It would be nice if SFSafariViewController had an option to open in private browsing mode like it does for reader mode.

**Comment by kolpanic** [comment]
There's [new API](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller/datastore/clearwebsitedata(completionhandler:)) to clear SFSafariViewController's cache. However, it became available in iOS 16 (which would require a bump of NNW's target from iOS 13).

For context, I've subscribed to the feed for [the Toronto Star's weekly roundup of restaurant inspections](http://www.thestar.com/search/?f=rss&t=article&c=news/gta/food-inspections&l=50&s=start_time&sd=desc), and their paywall allows only one article view per month without ponying up.
